File: src/test/java/SeleniumTests.java
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import dto.BuildType;
import dto.Builds;
import io.restassured.RestAssured;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Ignore;
import org.junit.Test;

import java.util.UUID;

@Ignore
public class SeleniumTests extends BaseWebTest {
    String adminUsername = "admin";
    String adminPassword = "admin1";
    String devUsername = "user1";
    String devPassword = "user1";
    TeamcityClient teamcityClientForDataPrepare;

    @Before
    public void initTeamcityApiClient() {
        RestAssured.baseURI = getBASE_URI();
        RestAssured.basePath = "/app/rest/";
        teamcityClientForDataPrepare = new TeamcityClient("admin", "admin1");
    }

    @Test
    public void testAvailabilityAdministrationModuleByDevUser() {
        LoginPage loginPage = new LoginPage(getDriver(), getBASE_URI());
        loginPage.open();
        loginPage.loginAs(devUsername, devPassword);
        Assert.assertEquals(true, new UserPanel(getDriver()).isAdministrationPageLinkAbset());
    }

    @Test
    public void testCreateNewUserByAdmin() {
        String testUser = UUID.randomUUID().toString();
        String testPassword = UUID.randomUUID().toString();
        LoginPage loginPage = new LoginPage(getDriver(), getBASE_URI());
        loginPage.open();
        loginPage.loginAs(adminUsername, adminPassword);
        AdministrationPage administrationPage = new UserPanel(getDriver()).openAdministrationPage();
        UsersModule userModuleBeforeAddingUser = administrationPage.openUsersModule();
        Long userCountBefore = userModuleBeforeAddingUser.getUsersCount();
        UserCreationPage userCreationPage = userModuleBeforeAddingUser.getUserCreationPage();
        userCreationPage.fillUserField(testUser);
        userCreationPage.fillPasswordField(testPassword);
        userCreationPage.fillRetypedPasswordField(testPassword);
        userCreationPage.clickSubmitButton();

        UsersModule userModuleAfterAddingUser = administrationPage.openUsersModule();
        Long expectedCount = userCountBefore + 1;
        Assert.assertEquals(expectedCount, userModuleAfterAddingUser.getUsersCount());
    }

    @Test
    public void incorrectEnterRetypePasswordOnUserCreation_shouldGiveError() {
        String testUser = UUID.randomUUID().toString();
        String testPassword = UUID.randomUUID().toString();
        LoginPage loginPage = new LoginPage(getDriver(), getBASE_URI());
        loginPage.open();
        loginPage.loginAs(adminUsername, adminPassword);
        AdministrationPage administrationPage = new UserPanel(getDriver()).openAdministrationPage();
        UsersModule userModuleBeforeAddingUser = administrationPage.openUsersModule();
        Long userCountBefore = userModuleBeforeAddingUser.getUsersCount();
        UserCreationPage userCreationPage = userModuleBeforeAddingUser.getUserCreationPage();
        userCreationPage.fillUserField(testUser);
        userCreationPage.fillPasswordField(testPassword);
        userCreationPage.fillRetypedPasswordField("incorrectPwd");
        userCreationPage.clickSubmitButton();

        Assert.assertEquals("Passwords mismatch", userCreationPage.tryLookForPasswordError());

        new UserPanel(getDriver()).openAdministrationPage();
        UsersModule userModuleAfterAddingUser = administrationPage.openUsersModule();
        Assert.assertEquals(userCountBefore, userModuleAfterAddingUser.getUsersCount());
    }

    @Test
    public void runBuildWithParameters() {
        BuildType testBuildType = teamcityClientForDataPrepare.createUniqueBuildType();
        String testBuildTypeName = testBuildType.getName();
        String testProjectName = testBuildType.getProjectName();
        LoginPage loginPage = new LoginPage(getDriver(), getBASE_URI());
        loginPage.open();
        OverviewPage overviewPage = loginPage.loginAs(adminUsername, adminPassword);
        CustomBuildDialog dialog = overviewPage.navigateToProjectByName(testProjectName).navigateToBuildTypeByName(testBuildTypeName).
                navigateToCustomBuildDialog();
        dialog.navigateToParameters().addNewConfigurationParameter("name", "value");
        dialog.submitRunBuild();
        Boolean d = teamcityClientForDataPrepare.getBuildQueue().getBuild().stream().
                anyMatch((x) -> x.getBuildTypeId().equals(testBuildType.getId()));
        Assert.assertEquals(true, d);
    }
}

File: src/test/kotlin/SeleniumTests.kt
import io.restassured.RestAssured
import org.junit.*
import pageModel.LoginPage
import pageModel.UserPanel

import java.util.UUID

class SeleniumTests : BaseUITest() {
    companion object {
        @BeforeClass
        @JvmStatic
        fun initTeamcityApiClient() {
            RestAssured.baseURI = TEAMCITY_BASE_URI_FOR_LOCAL
            RestAssured.basePath = "/app/rest/"
        }
    }

    @Test
    fun testAvailabilityAdministrationModuleByDevUser() {
        val loginPage = LoginPage(driver, TEAMCITY_BASE_URI_FOR_DRIVER)
        loginPage.open().loginAs(devUsername, devPassword)

        Assert.assertEquals(true, UserPanel(driver).isAdministrationPageLinkAbset)
    }

    @Test
    fun testCreateNewUserByAdmin() {
        val testUser = UUID.randomUUID().toString()
        val testPassword = UUID.randomUUID().toString()
        LoginPage(driver, TEAMCITY_BASE_URI_FOR_DRIVER).open().loginAs(adminUsername, adminPassword)

        val administrationPage = UserPanel(driver).openAdministrationPage()
        val userModuleBeforeAddingUser = administrationPage.openUsersModule()
        val userCountBefore = userModuleBeforeAddingUser.getUsersCount()
        val userCreationPage = userModuleBeforeAddingUser.getUserCreationPage()

        userCreationPage
                .fillUserField(testUser)
                .fillPasswordField(testPassword)
                .fillRetypedPasswordField(testPassword)
                .clickSubmitButton()

        val userModuleAfterAddingUser = administrationPage.openUsersModule()
        Assert.assertEquals(userCountBefore!! + 1, userModuleAfterAddingUser.getUsersCount())
    }

    @Test
    fun incorrectEnterRetypePasswordOnUserCreation_shouldGiveError() {
        val testUser = UUID.randomUUID().toString()
        val testPassword = UUID.randomUUID().toString()
        LoginPage(driver, TEAMCITY_BASE_URI_FOR_DRIVER).open().loginAs(adminUsername, adminPassword)

        val administrationPage = UserPanel(driver).openAdministrationPage()
        val userModuleBeforeAddingUser = administrationPage.openUsersModule()
        val userCountBefore = userModuleBeforeAddingUser.getUsersCount()
        val userCreationPage = userModuleBeforeAddingUser.getUserCreationPage()

        userCreationPage.fillUserField(testUser)
                .fillPasswordField(testPassword)
                .fillRetypedPasswordField("incorrectPwd")
                .clickSubmitButton()

        Assert.assertEquals("Passwords mismatch", userCreationPage.tryLookForPasswordError())

        val userModuleAfterAddingUser =  UserPanel(driver).openAdministrationPage().openUsersModule()
        Assert.assertEquals(userCountBefore, userModuleAfterAddingUser.getUsersCount())
    }

    @Test
    fun runBuildWithParameters() {
        val buildType = teamcityClientForDataPrepare.createUniqueBuildType()

        val overviewPage = LoginPage(driver, TEAMCITY_BASE_URI_FOR_DRIVER).open().loginAs(adminUsername, adminPassword)
        val dialog = overviewPage
                .navigateToProjectByName(buildType.projectName)
                .navigateToBuildTypeByName(buildType.name)
                .navigateToCustomBuildDialog()

        dialog.navigateToParameters().addNewConfigurationParameter("name", "value")
        dialog.submitRunBuild()

        val numberOfBuildsOfThisTypeInQueue = teamcityClientForDataPrepare.getBuildQueue().build.count { build -> build.buildTypeId == buildType.id }
        Assert.assertEquals("Number of builds of this buildType in queue was different of expected!", 1, numberOfBuildsOfThisTypeInQueue)
    }
}